timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
+        return jsonify({"objects": sigma_rules, "meta": meta})
+
+
+class SigmaRuleResource(resources.ResourceMixin, Resource):
+    """Resource to get / delete / post / put a Sigma rule."""
+
+    @login_required
+    def get(self, rule_uuid):
+        """Handles GET request to the Sigma Rule resource.
+
+        Handels GET API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            JSON sigma rule
+        """
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e:  # pylint: disable=broad-except
+            logger.error(
+                "Unable to get the Sigma rule",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "Unable to get Sigma rule. Error {0!s}".format(e),
+            )
+
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rules = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rules.append(_enrich_sigma_rule_object(rule=rule))
+
+        return jsonify({"objects": return_rules, "meta": {}})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Handles DELETE request to the resource.
+
+        Handels DELETE API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            HTTP_STATUS_CODE_NOT_FOUND if rule not found
+            HTTP_STATUS_CODE_OK if rule deleted
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        if not rule:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule found with rule_uuid.{0!s}".format(rule_uuid),
+            )
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+        Handels POST API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+        The remaining attributes of the rule are provided in request itself.
+
+        If no `rule_yaml` is found in the reuqest, the method will fail as this
+        is required to parse the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        rule_yaml = form.get("rule_yaml")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
```

Comment:
I don't think this should return a 404, rather a 400 https://en.wikipedia.org/wiki/List_of_HTTP_status_codes or maybe a 403. Whatever you settle on, it should still be the same error than on L333. Let's try to be consistent in the error code we return when the request is valid but the sigma rule isn't.